src/gui/MainWindow.cpp

```diff
@@ -379,16 +379,16 @@ void MainWindow::finalize()
 	m_toolsMenu = new QMenu( this );
 	for( const Plugin::Descriptor* desc : getPluginFactory()->descriptors(Plugin::Tool) )
 	{
-		m_toolsMenu->addAction( desc->logo->pixmap(), desc->displayName );
-		m_tools.push_back( ToolPlugin::instantiate( desc->name, /*this*/nullptr )
-						   ->createView(this) );
-	}
-	if( !m_toolsMenu->isEmpty() )
-	{
-		menuBar()->addMenu( m_toolsMenu )->setText( tr( "&Tools" ) );
-		connect( m_toolsMenu, SIGNAL(triggered(QAction*)),
-					this, SLOT(showTool(QAction*)));
+		const auto objectName = QString("ToolsMenuAction") + desc->displayName;
```

Comment:
I wonder why you use `displayName` while there is also `name`. Plugins are instantiated using `name`, so I guess `name` is more natural here.

Reply:
@irrenhaus3 Friendly reminder that this has an open comment.